tests/tape/test_tape.py

```diff
@@ -806,7 +806,7 @@ def test_array_parameter(self):
         assert tape.num_params == len(params)
         assert tape.get_parameters() == params
 
-        b = np.array([0, 1, 0, 0])
+        b = np.array([0.0, 1.0, 0.0, 0.0])
```

Comment:
What happens when one uses vectors with `dtype=int`? 🤔

Reply:
I think I'll fix it with this last commit

The problem is that now I preprocess in StatePrep and it is transforming it by default into a float array

Reply:
Hmm didn't fix it, doesn't seem like an easy thing to avoid